docs/ref/templates/builtins.txt

```diff
@@ -38,6 +29,38 @@ Sample usage:
         {{ body }}
     {% endautoescape %}
 
+When auto-escaping is in effect, all content derived from variables have HTML
+escaping applied to it before placing the result into the output (but after any
+filters have been applied). This is equivalent to manually applying the
+:tfilter:`escape` filter to each variable.
+
+The only exceptions are variables that are already marked as "safe" from
+escaping: either by the code that populated the variable, or because it has had
+the :tfilter:`safe` or :tfilter:`escape` filters applied, or because it's the
+result of a previous filter that marked the string as "safe".
+
+It's important to note that within the scope of disabled auto-escaping,
+chaining filters including :tfilter:`escape` could lead to unexpected (but
+documented) results such as the following:
+
+.. code-block:: html+django
+
+    {% autoescape off %}
+        {{ my_list|join:", "|escape }}
+    {% endautoescape %}
+
+The above code will output the joined elements of ``my_list`` unescaped. This
+is because the filter chaining sequence executes first :tfilter:`join` on
+``my_list`` (without applying escaping to each item since ``autoescape`` is
+``off``), marking the result as safe. Subsequently, this safe result will be
+fed to :tfilter:`escape` filter, which, as documented, does not apply a second
+round of escaping.
+
+..
+    Once https://code.djangoproject.com/ticket/34577 gets released, we should
+    add a sentence explaining how using :tfilter:`escapeseq` could be used to
+    properly implement the example above"
```

Comment:
This is a comment to remember to extend this section.

Reply:
Can we add `escapeseq` first? :thinking:

Reply:
Sure, updating now that the filter was added.